### PR TITLE
JAMES-3601 Piggy back content cleaning upon browse start update

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 1]
```diff
@@ -288,6 +292,51 @@ void browseAndDequeueShouldCombineWellWhenDifferentSlices() throws Exception {
...
+            clock.setInstant(IN_SLICE_7);
+            dequeueMails(5);
+            dequeueMails(5);
+            dequeueMails(5);
+            dequeueMails(5);
+
+            // ensure slice 1 was cleaned
+            EnqueuedMailsDAO mailsDAO = new EnqueuedMailsDAO(cassandraCluster.getCassandraCluster().getConf(), new HashBlobId.Factory());
+            MailQueueName queueName = MailQueueName.fromString(mailQueue.getName().asString());
+            Slice slice = Slice.of(Instant.parse("2007-12-03T10:15:00.00Z")).allSlicesTill(IN_SLICE_2, ONE_HOUR_SLICE_WINDOW).findFirst()
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java:321]: Not really, the slice need to be the exact instant

**Comment by chibenwa** [comment]
Blobs are already deleted at dequeue time...

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebased and squash fixups

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/DeletedMailsDAO.java [hunk 2]
```diff
@@ -65,6 +68,13 @@ private PreparedStatement prepareSelectExist(Session session) {
             .and(eq(ENQUEUE_ID, bindMarker(ENQUEUE_ID))));
     }
 
+    private PreparedStatement prepareDeleteOne(Session session) {
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/DeletedMailsDAO.java:71]: aren't you afraid of tombstone?

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java [hunk 3]
```diff
@@ -131,6 +131,14 @@ public CassandraMailQueueItemView next() {
             .flatMapSequential(slice -> browseSlice(queueName, slice));
     }
 
+    Flux<EnqueuedItemWithSlicingContext> browseReferences(MailQueueName queueName, Instant start, Instant end) {
+        return allSlicesStartingAt(start)
+            .doOnNext(n -> System.out.println("     --- " + n + " slice looked up"))
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java:136]: println?

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/model/BucketedSlices.java [hunk 4]
```diff
@@ -119,5 +119,10 @@ public final boolean equals(Object o) {
         public final int hashCode() {
             return Objects.hash(startSliceInstant);
         }
+
+        @Override
+        public String toString() {
+            return "" + startSliceInstant;
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/model/BucketedSlices.java:125]: why use a concat to do a toString?

**Review by mbaechler — APPROVED** [review]

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/DeletedMailsDAO.java (hunk 2 above)
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/DeletedMailsDAO.java:71]: No, we don't care as we tombstone data that will never ever be read again :-)

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java (hunk 3 above)
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java:136]: @Arsnael can you fix it up?

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/model/BucketedSlices.java (hunk 4 above)
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/model/BucketedSlices.java:125]: This is a debug call, we can use MoreObjects here (@Arsnael  maybe?)

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java (hunk 3 above)
Comment (Arsnael) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: I fixed it, but I do have a concern as I had my hands directly in the code. This newly created method does not seem to be used anywhere. So did you forget to switch to it from pieces of code using the other `browseReferences` method? Or is this useless and should be removed?

Comment (Arsnael) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: cc @chibenwa

Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: True I moved away from it.

We can remove it.

Rationnals: as items prior browse start are all marked as deleted, then browse is not returning them. I had to use another strategy.

Please @Arsnael can you remove it?

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Squashed fixups

**Pull request merged**